src/wp-includes/link-template.php

```diff
@@ -262,7 +262,10 @@ function get_permalink( $post = 0, $leavename = false ) {
 		$author = '';
 		if ( strpos( $permalink, '%author%' ) !== false ) {
 			$authordata = get_userdata( $post->post_author );
-			$author     = $authordata->user_nicename;
+
+			if ( false !== $authordata ) {
```

Comment:
```suggestion
			if ( $authordata instanceof WP_User ) {
```

Always better to verify that the type if the type want, rather than comparing against one of the many types you don't want.

Would also be good to see a test covering this change.